src/amplifyp/gui/views/input/primer_action_controller.py

```
@@ -190,16 +190,32 @@ def move_primers(self, indices: set[int], direction: int) -> None:
             self.owner.on_change_handler(None)
 
     def delete_primers(self, indices_to_delete: set[int]) -> None:
-        """Delete primers at indices and re-index controls in-place.
+        """Delete primers asynchronously to avoid Flet race conditions."""
+        page = None
+        try:
+            page = self.owner.page
+        except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)